Fix -z now is really a LD option, not a gcc option so use -Wl to pass it to LD

Created-by: Saul Wold <saul.wold@intel.com>

--- Makefile.orig	2010-08-31 13:52:46.277764099 -0700
+++ Makefile	2010-08-31 14:02:28.218014078 -0700
@@ -167,7 +167,7 @@
 
 # use '-z now' to guarantee no dynamic linker interactions with the monitor thread
 mdmon : $(MON_OBJS)
-	$(CC) $(LDFLAGS) $(MON_LDFLAGS) -z now -o mdmon $(MON_OBJS) $(LDLIBS)
+	$(CC) $(LDFLAGS) $(MON_LDFLAGS) -Wl,-z,now -o mdmon $(MON_OBJS) $(LDLIBS)
 msg.o: msg.c msg.h
 
 test_stripe : restripe.c mdadm.h
